### PR TITLE
new: type inference, upgrading to ESmodule and cleaning up

### PR DESCRIPTION
**Pull request by ItamarShDev**
### Description

In this PR, I have made the following changes:
1. Upgraded the ts file to ES6 and target to node.
2. The API will return JSON and throw an error on failure, to be compatible with best practices.
3. Added type inference for the api calls. if the conversion fails - the api returns a JSON object
4. cleaned up the code to handle errors easier
### Checklist

- [x] Code compiles correctly
- [x] Created tests which fail without the change (if applicable)
- [x] All lint and unit tests passing
- [x] Extended the README / documentation, if necessary

### Additional Comments

Anything else you'd like to add.

### CODE REVIEW

**Comment by bgrnwd** [comment]
@ItamarShDev Thanks for the PR, give me a couple days to look it over.

**Comment by ItamarShDev** [comment]
> @ItamarShDev Thanks for the PR, give me a couple days to look it over.

Thanks!

File: package.json [hunk 1]
```diff
@@ -1,5 +1,4 @@
 {
-  "name": "fotmob",
   "version": "1.1.0",
```
Comment (bgrnwd) [inline package.json:2]: Mind bumping this to `2.0.0`?

File: test/fotmob.test.ts
Comment (bgrnwd) [inline test/fotmob.test.ts:1]: Would you mind adding some additional test cases to make sure the new types are working correctly?

**Review by bgrnwd — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -1,5 +1,4 @@
 {
-  "name": "fotmob",
```
[the remark this answers is not in the document]
Comment (ItamarShDev) [reply package.json:2]: Mistake, adding back

File: package.json (hunk 1 above)
Comment (ItamarShDev) [reply package.json:2]: done

File: test/fotmob.test.ts
Comment (ItamarShDev) [reply test/fotmob.test.ts:1]: Added

**Review by bgrnwd — APPROVED** [review]
Thanks for the contribution! I'll get this merged in and a new release out there shortly

**Comment by ItamarShDev** [comment]
> Thanks for the contribution! I'll get this merged in and a new release out there shortly

Thanks! patiently waiting for the version :)

**Comment by bgrnwd** [comment]
It should be out there now!

**Pull request merged**